core/synch.c

```diff
@@ -1815,6 +1815,8 @@ translate_from_synchall_to_dispatch(thread_record_t *tr, thread_synch_state_t sy
          */
         mc->pc = (app_pc)get_reset_exit_stub(dcontext);
         LOG(GLOBAL, LOG_CACHE, 2, "\tsent to reset exit stub " PFX "\n", mc->pc);
+        /* TODO i#4497: Replace with the official fix from PR#4498. */
```

Comment:
Well, the line below is already the same fix used in PR#4498 too.

Reply:
I figured PR #4498 would go in before this one.  Though at this point maybe not.  Are you suggesting to remove this fix if the tests on the tx1 and jenkins don't hit it and assume PR #4498 will be in before anyone happens to hit it on the new test on a new platform?

Reply:
PR #4498 is currently blocked on #4496 (found another `ASSERT_NOT_IMPLEMENTED` for AArch64 after the `save_fpstate` one).

It should be okay to include the fix in this PR if it helps avoid anyone else hitting and debugging it. Though, do add an `ASSERT_NOT_TESTED` for #4498 work on improving the tests to hit this path.

Reply:
OK, I'm ok either way.  I'll leave this line plus an ASSERT_NOT_TESTED as requested.

Reply:
Would be good to xref `Issue #4497` in the final commit description.

Reply:
Will do.